<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Štěpánka Pustková, Branislav Kopčan">

<title>gordi: Plot ordination diagrams with ggplot2 – Botzool Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-0508618caa1d256af916c83e0797e277.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar docked quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../CommunityEcology/community_ecology.html">Community Ecology</a></li><li class="breadcrumb-item"><a href="../CommunityEcology/gordi_vignette.html">gordi</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../ubz_logo_eng_2barvy.jpg" alt="Botzool Logo" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../DataManipulationVisualisation/data_manipulation_visualisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Manipulation and Visualization</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/1_introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1 Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/2_data_manipulation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2 Data manipulation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/3_data_visualisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3 Data visualisation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/4_wide_vs_long.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4 Wide vs.&nbsp;long format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/5_join_functions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5 Join functions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/6_advanced_visualisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6 Advanced data visualisation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/7_8_automatisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7 + 8 Automatisation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/9_10_maps.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9 + 10 Maps</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/11_database_to_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11 From a database to a plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataManipulationVisualisation/12_github.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 GitHub</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../CommunityEcology/community_ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Community Ecology</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../CommunityEcology/gordi_vignette.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">gordi</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../AIinTeaching/ai_in_teaching.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">AI in Data Analysis</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#general-workflow" id="toc-general-workflow" class="nav-link" data-scroll-target="#general-workflow">General Workflow</a>
  <ul class="collapse">
  <li><a href="#calculate-ordination" id="toc-calculate-ordination" class="nav-link" data-scroll-target="#calculate-ordination">1. Calculate ordination</a></li>
  <li><a href="#read-the-ordination-result" id="toc-read-the-ordination-result" class="nav-link" data-scroll-target="#read-the-ordination-result">2. Read the ordination result</a></li>
  <li><a href="#choose-what-to-display" id="toc-choose-what-to-display" class="nav-link" data-scroll-target="#choose-what-to-display">3. Choose, what to display</a></li>
  <li><a href="#change-aestethics" id="toc-change-aestethics" class="nav-link" data-scroll-target="#change-aestethics">4. Change aestethics</a></li>
  <li><a href="#add-predictors" id="toc-add-predictors" class="nav-link" data-scroll-target="#add-predictors">5. Add predictors</a></li>
  <li><a href="#customise-the-plot" id="toc-customise-the-plot" class="nav-link" data-scroll-target="#customise-the-plot">6. Customise the plot</a></li>
  </ul></li>
  <li><a href="#list-of-gordi_-functions" id="toc-list-of-gordi_-functions" class="nav-link" data-scroll-target="#list-of-gordi_-functions">List of <code>gordi_</code> functions</a></li>
  </ul></li>
  <li><a href="#limitations" id="toc-limitations" class="nav-link" data-scroll-target="#limitations">Limitations</a></li>
  <li><a href="#contact-the-authors" id="toc-contact-the-authors" class="nav-link" data-scroll-target="#contact-the-authors">Contact the authors</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../CommunityEcology/community_ecology.html">Community Ecology</a></li><li class="breadcrumb-item"><a href="../CommunityEcology/gordi_vignette.html">gordi</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">gordi: Plot ordination diagrams with ggplot2</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Štěpánka Pustková, Branislav Kopčan </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<p><img src="images/clipboard-727467331.png" class="img-fluid"> Logo by Saša Smolková</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gordi)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>The <code>gordi</code> package was developed to help students of community ecology deal with community data analysis, especially when it comes to <strong>ordination methods</strong> and the visualisation of their results in R.</p>
<p>With <code>gordi</code>, you can:</p>
<ul>
<li><p>take results from common ordinations in the <strong>vegan</strong> package, particularly:</p>
<ul>
<li><code>rda()</code> for PCA and RDA</li>
<li><code>cca()</code> for CA and CCA</li>
<li><code>capscale()</code> for PCoA and db-RDA</li>
<li><code>metaMDS()</code> for NMDS</li>
<li><code>decorana()</code> for DCA</li>
</ul></li>
<li><p>visualise them quickly with clear and customisable plots via the <strong>ggplot2</strong> framework</p></li>
<li><p>and have access to <strong>ggplot2</strong> for final polishing</p></li>
</ul>
<p><code>gordi</code> is still under active development, so changes happen frequently. We recommend updating the package often to get the latest features. Slight differences in results between versions may occur due to these updates. <code>gordi</code> is currently hosted on Github as it will take some time (and many bug fixes) before it will be prepared for publication on CRAN (the main R package repository). It can be installed (or updated) using this code:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">'Kryshtoson/gordi'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="general-workflow" class="level2">
<h2 class="anchored" data-anchor-id="general-workflow">General Workflow</h2>
<p>The functions in <code>gordi</code> follow some rules regarding the order in which they are called. The most important rule is, that every <code>gordi</code> workflow must start with the <code>gordi_read()</code> function. Second, <code>gordi_</code> functions work the best if applied in a pipeline, as the output (<code>pass</code>) of one function becomes the input for the next. The pipeline approach keeps your code clean and better readable.</p>
<p>The ordination analysis as a whole has basically three parts. The first involves data preparation. It usually consists of synchronising taxonomical concepts in the dataset, fixing mistakes, typos, deciding what to do with outlier samples, and finally, putting the data into a format, that is acceptable by the analytical software.</p>
<p>In the second part, the model is typically calculated. It tends to be the shortest, but the most important one.</p>
<p>Only in the third part, the visualisation of results, does <code>gordi</code> step in. In this vignette, we will cover the second (slightly) and predominantly the third part. The first part (data preparation) is omitted here, as it is usually highly case specific, but we will mention the format of the data which <code>gordi</code> needs to function correctly.</p>
<section id="calculate-ordination" class="level3">
<h3 class="anchored" data-anchor-id="calculate-ordination">1. Calculate ordination</h3>
<p>The first step is calculating an ordination. The <code>gordi</code> package is compatible with the outputs of <code>vegan</code> functions mentioned above.</p>
<p>In this example, we will use the <code>dune</code> dataset which is available in the library <code>vegan</code> to calculate a simple constrained ordination (db-RDA). We will square-root the species data (often called <strong>spe</strong>) to reduce the influence of dominant species. As the environmental predictor, we will use the <code>A1</code> variable which is available in the environmental table (often called <strong>env</strong>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gordi)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Load example dataset</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(dune)        <span class="co"># spe data</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"dune.env"</span>)  <span class="co"># env data</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Check if the data have the correct format</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># e.g. species table must not contain any ID column,</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># the number of rows in spe and env table must be the same</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(dune)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 20
Columns: 30
$ Achimill &lt;dbl&gt; 1, 3, 0, 0, 2, 2, 2, 0, 0, 4, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0
$ Agrostol &lt;dbl&gt; 0, 0, 4, 8, 0, 0, 0, 4, 3, 0, 0, 4, 5, 4, 4, 7, 0, 0, 0, 5
$ Airaprae &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 0, 3, 0
$ Alopgeni &lt;dbl&gt; 0, 2, 7, 2, 0, 0, 0, 5, 3, 0, 0, 8, 5, 0, 0, 4, 0, 0, 0, 0
$ Anthodor &lt;dbl&gt; 0, 0, 0, 0, 4, 3, 2, 0, 0, 4, 0, 0, 0, 0, 0, 0, 4, 0, 4, 0
$ Bellpere &lt;dbl&gt; 0, 3, 2, 2, 2, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0
$ Bromhord &lt;dbl&gt; 0, 4, 0, 3, 2, 0, 2, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
$ Chenalbu &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0
$ Cirsarve &lt;dbl&gt; 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
$ Comapalu &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 2, 0, 0, 0, 0, 0
$ Eleopalu &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 4, 5, 8, 0, 0, 0, 4
$ Elymrepe &lt;dbl&gt; 4, 4, 4, 4, 4, 0, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
$ Empenigr &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 0
$ Hyporadi &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 2, 0, 5, 0
$ Juncarti &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 4, 4, 0, 0, 0, 0, 0, 3, 3, 0, 0, 0, 4
$ Juncbufo &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 2, 0, 4, 0, 0, 4, 3, 0, 0, 0, 0, 0, 0, 0
$ Lolipere &lt;dbl&gt; 7, 5, 6, 5, 2, 6, 6, 4, 2, 6, 7, 0, 0, 0, 0, 0, 0, 2, 0, 0
$ Planlanc &lt;dbl&gt; 0, 0, 0, 0, 5, 5, 5, 0, 0, 3, 3, 0, 0, 0, 0, 0, 2, 3, 0, 0
$ Poaprat  &lt;dbl&gt; 4, 4, 5, 4, 2, 3, 4, 4, 4, 4, 4, 0, 2, 0, 0, 0, 1, 3, 0, 0
$ Poatriv  &lt;dbl&gt; 2, 7, 6, 5, 6, 4, 5, 4, 5, 4, 0, 4, 9, 0, 0, 2, 0, 0, 0, 0
$ Ranuflam &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 2, 2, 2, 2, 0, 0, 0, 4
$ Rumeacet &lt;dbl&gt; 0, 0, 0, 0, 5, 6, 3, 0, 2, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0
$ Sagiproc &lt;dbl&gt; 0, 0, 0, 5, 0, 0, 0, 2, 2, 0, 2, 4, 2, 0, 0, 0, 0, 0, 3, 0
$ Salirepe &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 3, 5
$ Scorautu &lt;dbl&gt; 0, 5, 2, 2, 3, 3, 3, 3, 2, 3, 5, 2, 2, 2, 2, 0, 2, 5, 6, 2
$ Trifprat &lt;dbl&gt; 0, 0, 0, 0, 2, 5, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
$ Trifrepe &lt;dbl&gt; 0, 5, 2, 1, 2, 5, 2, 2, 3, 6, 3, 3, 2, 6, 1, 0, 0, 2, 2, 0
$ Vicilath &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, 0, 0, 0, 0, 0, 0, 1, 0, 0
$ Bracruta &lt;dbl&gt; 0, 0, 2, 2, 2, 6, 2, 2, 2, 2, 4, 4, 0, 0, 4, 4, 0, 6, 3, 4
$ Callcusp &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 3, 0, 0, 0, 3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(dune.env)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 20
Columns: 5
$ A1         &lt;dbl&gt; 2.8, 3.5, 4.3, 4.2, 6.3, 4.3, 2.8, 4.2, 3.7, 3.3, 3.5, 5.8,…
$ Moisture   &lt;ord&gt; 1, 1, 2, 2, 1, 1, 1, 5, 4, 2, 1, 4, 5, 5, 5, 5, 2, 1, 5, 5
$ Management &lt;fct&gt; SF, BF, SF, SF, HF, HF, HF, HF, HF, BF, BF, SF, SF, NM, NM,…
$ Use        &lt;ord&gt; Haypastu, Haypastu, Haypastu, Haypastu, Hayfield, Haypastu,…
$ Manure     &lt;ord&gt; 4, 2, 4, 4, 2, 2, 3, 3, 1, 1, 1, 2, 3, 0, 0, 3, 0, 0, 0, 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dune) <span class="sc">==</span> <span class="fu">nrow</span>(dune.env)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Calculate the db-RDA ordination</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">capscale</span>(<span class="fu">sqrt</span>(dune) <span class="sc">~</span> A1,    </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">data =</span> dune.env,    <span class="co"># we have to provide env table with predictors,</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="co"># so the model knows where to look for the predictor</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">distance =</span> <span class="st">'bray'</span>,  <span class="co"># 'Bray-Curtis dissimilarity' - a common choice for community data</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">sqrt.dist =</span> T)      <span class="co"># square-rooted Bray-Curtis dissim. does not produce</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="co"># Imaginary Inertia (negative eigenvalues),</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="co"># which is problematic, when it comes to determining</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="co"># how much variability in the data the model explained</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-the-ordination-result" class="level3">
<h3 class="anchored" data-anchor-id="read-the-ordination-result">2. Read the ordination result</h3>
<p>Now it is time for the first <code>gordi</code> function. <code>gordi_read()</code> reads the resulting object of the ordination (<strong>m</strong>) and prepares it for use by other <code>gordi_</code> functions. That said, <code>gordi_read()</code> has to always be the first in the pipeline of <code>gordi_</code> functions.</p>
<p><code>gordi_read()</code> has many arguments, which greatly influence how the final plot will look like. You can set here the scaling of the plot, select the ordination axes you want to look at. You can also provide <strong>env</strong>, <strong>spe</strong> and <strong>traits</strong> data which you can later use for variable based symbolisation of sites or species in the plot or for passive projection of variables onto unconstrained ordination.</p>
<p>For more details regarding the arguments used in the <code>gordi_read()</code> please check the help page of the <code>scores()</code> function in the <code>vegan</code> package.</p>
<blockquote class="blockquote">
<p><strong>Note</strong>: If you check the help page for “scores”, it will be rather laconic. For more details on the arguments and how do they influence the calculation of scores, you have to look for the ordination type specific <code>scores()</code> function, such as “scores.rda” or “scores.cca”.</p>
</blockquote>
<blockquote class="blockquote">
<p><strong>Note</strong>: Don’t panic when you see the large output printed to the Console. This is the <code>pass</code> object which travels through the pipeline of <code>gordi_</code> functions carrying all the necessary information needed for plotting. You can easily check if you entered the right data.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Print ordination result</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>m</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call: capscale(formula = sqrt(dune) ~ A1, data = dune.env, distance =
"bray", sqrt.dist = T)

              Inertia Proportion Rank
Total           5.699      1.000     
Constrained     0.678      0.119    1
Unconstrained   5.021      0.881   18

Inertia is Bray distance

-- NOTE:
Species scores projected from 'sqrt'
Species scores projected from 'dune'

Eigenvalues for constrained axes:
 CAP1 
0.678 

Eigenvalues for unconstrained axes:
  MDS1   MDS2   MDS3   MDS4   MDS5   MDS6   MDS7   MDS8 
1.0664 0.8972 0.5421 0.4314 0.4050 0.2744 0.2453 0.1891 
(Showing 8 of 18 unconstrained eigenvalues)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use `gordi_read()` to read the ordination result</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_read</span>(</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,         <span class="co"># we want to look at the first two ordination axes</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">scaling =</span> <span class="st">'species'</span>,   <span class="co"># we want to focus on the species</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">correlation =</span> T</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$m

Call: capscale(formula = sqrt(dune) ~ A1, data = dune.env, distance =
"bray", sqrt.dist = T)

              Inertia Proportion Rank
Total           5.699      1.000     
Constrained     0.678      0.119    1
Unconstrained   5.021      0.881   18

Inertia is Bray distance

-- NOTE:
Species scores projected from 'sqrt'
Species scores projected from 'dune'

Eigenvalues for constrained axes:
 CAP1 
0.678 

Eigenvalues for unconstrained axes:
  MDS1   MDS2   MDS3   MDS4   MDS5   MDS6   MDS7   MDS8 
1.0664 0.8972 0.5421 0.4314 0.4050 0.2744 0.2453 0.1891 
(Showing 8 of 18 unconstrained eigenvalues)


$type
[1] "db-RDA"

$explained_variation
      CAP1       MDS1       MDS2       MDS3       MDS4       MDS5       MDS6 
0.11897706 0.18713012 0.15743892 0.09512279 0.07570342 0.07107262 0.04815021 
      MDS7       MDS8       MDS9      MDS10      MDS11      MDS12      MDS13 
0.04304449 0.03318727 0.02994104 0.02652510 0.02343062 0.01907886 0.01792983 
     MDS14      MDS15      MDS16      MDS17      MDS18 
0.01486430 0.01220752 0.01074280 0.00844054 0.00701251 

$choices
[1] 1 2

$scaling
[1] "species"

$correlation
[1] TRUE

$hill
[1] FALSE

$const
[1] 2 2

$axis_names
[1] "CAP1" "MDS1"

$site_scores
# A tibble: 20 × 2
      CAP1      MDS1
     &lt;dbl&gt;     &lt;dbl&gt;
 1 -0.756  -0.0517  
 2 -0.563  -0.208   
 3 -0.211   0.143   
 4 -0.263   0.150   
 5 -0.505  -0.818   
 6 -0.454  -0.552   
 7 -0.559  -0.367   
 8  0.287   0.568   
 9 -0.0969  0.397   
10 -0.590  -0.467   
11 -0.426  -0.197   
12  0.295   0.301   
13  0.318   0.313   
14  1.23   -0.000327
15  1.24   -0.124   
16  0.867   0.814   
17 -0.326  -0.476   
18 -0.312  -0.352   
19 -0.0791  0.00428 
20  0.905   0.922   

$species_scores
# A tibble: 30 × 2
       CAP1    MDS1
      &lt;dbl&gt;   &lt;dbl&gt;
 1 -0.0908  -0.157 
 2  0.114    0.180 
 3 -0.0431  -0.0358
 4 -0.00467  0.131 
 5 -0.0600  -0.148 
 6 -0.0430  -0.0854
 7 -0.0662  -0.0937
 8  0.0338   0.0367
 9 -0.0191   0.0176
10  0.237   -0.0106
# ℹ 20 more rows

$predictor_scores
# A tibble: 1 × 4
   CAP1  MDS1 score  label
  &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;  &lt;chr&gt;
1     1     0 biplot A1   

$species_names
# A tibble: 30 × 1
   species_names
   &lt;chr&gt;        
 1 Achimill     
 2 Agrostol     
 3 Airaprae     
 4 Alopgeni     
 5 Anthodor     
 6 Bellpere     
 7 Bromhord     
 8 Chenalbu     
 9 Cirsarve     
10 Comapalu     
# ℹ 20 more rows

$predictor_names
# A tibble: 1 × 1
  predictor_names
  &lt;chr&gt;          
1 A1             

$env
NULL

$spe
NULL

$traits
NULL</code></pre>
</div>
</div>
</section>
<section id="choose-what-to-display" class="level3">
<h3 class="anchored" data-anchor-id="choose-what-to-display">3. Choose, what to display</h3>
<p>At this point, we actually start plotting. The first decision often is, whether we want to display ‘species’ or ‘sites’ (samples) scores. In the previous example we decided, we want to focus on ‘species’, so we set the <code>scaling = 'species'</code>.</p>
<p>The scaling parameter determines how the <strong>scores</strong> are calculated. If we set <code>scaling = 'species'</code>, the plot is optimised in a way that species relationships are displayed correctly (i.e., arrows pointing in similar directions or close together means the species co-occur or respond similarly). This applies also for the situation when we want to look at ‘sites’ (samples) and we set <code>scaling = 'sites'</code>.</p>
<blockquote class="blockquote">
<p><strong>Note</strong>: The word “scores” (or “ordination scores”) might sound unfamiliar, but scores are simply the “coordinates” of each species or site (sample) on the ordination axis. They represent where that site (sample) or species falls in the multidimensional space that the ordination creates.</p>
</blockquote>
<p>On the other hand, if we set <code>scaling = 'sites'</code> and then we display ‘species’, their relative position will be harder to interpret. It still might be useful to look at both scores (sites and species) in one plot, but we have to keep on mind, that only what we scaled for will be displayed correctly.</p>
<p>There is also a third, compromising option how to display both, species and site (sample) scores at once, and this is <code>scaling = 'symmetric'</code>. The scores of both, species and sites, are moderately distorted (‘re-scaled’) but not that much as when you scale for species and display sites (and vice versa).</p>
<p>So, this was about scaling, but to actually display sites or species, we have to add another <code>gordi_</code> function to the pipeline:</p>
<ul>
<li>If you have decided to display species, use <code>gordi_species()</code></li>
<li>If you have decided to display sites, use <code>gordi_sites()</code></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_read</span>(<span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">scaling =</span> <span class="st">'species'</span>, <span class="at">correlation =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>OK, good. We have a bunch of blue arrows. Each arrow represent one species, but which one? For a quick information, we can set argument <code>label = T</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_read</span>(<span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">scaling =</span> <span class="st">'species'</span>, <span class="at">correlation =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">label =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Nice. We can see, that the names of species in the <code>dune</code> data set are already shortened, so we have to do nothing more about it. In case, when the data set is not that nicely prepared and you have the whole species names in the column names in the spe table (e.g.&nbsp;<em>Achillea millefolium agg.</em>), the plot gets crowded rapidly. For this purposes, there is a handy function <code>gordi_label()</code> which can create nice species names shortcuts.</p>
<blockquote class="blockquote">
<p><strong>Note:</strong> Another solution for an overcrowded plot is the <code>gordi_fit()</code> function which filters out less frequent species or species which have weak correlation with the displayed ordination axes. Check its help page.</p>
</blockquote>
<p>What else can we improve is the position of the species names, so they do not overlap each other. For this, set the argument <code>repel_label = T</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_read</span>(<span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">scaling =</span> <span class="st">'species'</span>, <span class="at">correlation =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">label =</span> T, <span class="at">repel_label =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can also decide, that arrows will look better in ‘hotpink3’.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_read</span>(<span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">scaling =</span> <span class="st">'species'</span>, <span class="at">correlation =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">label =</span> T, <span class="at">repel_label =</span> T, <span class="at">colour =</span> <span class="st">'hotpink3'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="change-aestethics" class="level3">
<h3 class="anchored" data-anchor-id="change-aestethics">4. Change aestethics</h3>
<p>In <code>gordi_species()</code> and <code>gordi_sites()</code> we can map colour, fill, shape, and many more things with regard to some variable (‘dynamically’). Instead of setting colour = ‘hotpink3’, we will set <code>colour = 'Subclass'</code>. This will give different colours to species from different Subclasses.</p>
<blockquote class="blockquote">
<p><strong>Note:</strong> For species, this variable (<code>Subclass</code>) must be supplied to the <code>traits</code> argument as a tibble or data frame. This table has a strict format: the <strong>first column must contain species names</strong> that exactly match the column names in the <strong>spe</strong> table. In every following column, every species must have assigned only one value.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># At first, we need to prepare traits table to be compatible with species table.</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The first column in the traits table must contain species names as used in species table</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">'dune.taxon'</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(dune.taxon, <span class="at">rownames =</span> <span class="st">'species'</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Now we continue with the gordi pipeline</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> <span class="fu">gordi_read</span>(</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">env =</span> dune.env,      <span class="co"># table with environmental variable</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">traits =</span> traits,     <span class="co"># table with traits, respective in this case,</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># table with species names, families and </span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># other stuff is used</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,       <span class="co"># choose 1st and 2nd ordination axes</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">correlation =</span> T,</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">scaling =</span> <span class="st">'species'</span>  <span class="co"># we want to display species, we choose scaling = 'species'</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span>  </span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">'Subclass'</span>,     <span class="co"># colour may be set statically,</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>                             <span class="co"># or dynamically, by using name of some</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>                             <span class="co"># variable in traits data frame</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.8</span>,             <span class="co"># opacity of symbols</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrow_size =</span> <span class="fl">0.5</span>)        <span class="co"># changes arrow size (in cm)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>However, these two functions allows you to use only the default colour palettes or shapes.</p>
<p>If we want to select some predefined or custom colour scale or different shapes, we have to do it immediately after <code>gordi_species()</code> or <code>gordi_sites()</code> with <code>gordi_colour()</code> and <code>gordi_shape()</code> functions. - <code>gordi_colour()</code> controls both, colour and fill (by setting argument <code>fill = T/F</code>) - <code>gordi_shape()</code> controls point shape (for details about the numbers of shape, see <code>?pch</code>)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> <span class="fu">gordi_read</span>(<span class="at">env =</span> dune.env,      </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">traits =</span> traits,   </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> T,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">scaling =</span> <span class="st">'species'</span>) <span class="sc">|&gt;</span>  </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">colour =</span> <span class="st">'Subclass'</span>,     </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha =</span> <span class="fl">0.8</span>,             </span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">arrow_size =</span> <span class="fl">0.5</span>) <span class="sc">|&gt;</span>       </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_colour</span>(<span class="at">scale =</span> <span class="st">'discrete'</span>,       <span class="co"># first, you must define the scale (see </span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>                                         <span class="co"># ?gordi_colour() for options)</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">family =</span> <span class="st">'brewer'</span>)        <span class="co"># then you must select from a family of colour scales or set one manually </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We may want to set the colours manually. This is also possible with <code>gordi_colour()</code> (or <code>gordi_shape()</code>) and in this case, we set the first two arguments like this: <code>scale = 'discrete</code>, <code>family = 'manual</code>, and then we have to define <code>values = c('lightsalmon3', 'palegreen2')</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> <span class="fu">gordi_read</span>(<span class="at">env =</span> dune.env,      </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">traits =</span> traits,   </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> T,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">scaling =</span> <span class="st">'species'</span>) <span class="sc">|&gt;</span>  </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">colour =</span> <span class="st">'Subclass'</span>,     </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha =</span> <span class="fl">0.8</span>,             </span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">arrow_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">linewidth =</span> <span class="fl">0.5</span>) <span class="sc">|&gt;</span>       </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_colour</span>(<span class="at">scale =</span> <span class="st">'discrete'</span>,       <span class="co"># first, you must define the scale (see </span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>                                         <span class="co"># ?gordi_colour() for options)</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">family =</span> <span class="st">'manual'</span>,        </span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">values =</span> <span class="fu">c</span>(<span class="st">'lightsalmon3'</span>, <span class="st">'palegreen2'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="add-predictors" class="level3">
<h3 class="anchored" data-anchor-id="add-predictors">5. Add predictors</h3>
<p>At the beginning, we calculated constrained ordination (<code>RDA</code>). Predictors also have their scores and we will now use <code>gordi_predict()</code> to add them to the plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> <span class="fu">gordi_read</span>(<span class="at">env =</span> dune.env,      </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">traits =</span> traits,   </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> T,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">scaling =</span> <span class="st">'species'</span>) <span class="sc">|&gt;</span>  </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">colour =</span> <span class="st">'Subclass'</span>,     </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha =</span> <span class="fl">0.8</span>,             </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">arrow_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">linewidth =</span> <span class="fl">0.5</span>) <span class="sc">|&gt;</span>       </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_colour</span>(<span class="at">scale =</span> <span class="st">'discrete'</span>,    </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">family =</span> <span class="st">'manual'</span>,        </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">values =</span> <span class="fu">c</span>(<span class="st">'lightsalmon3'</span>, <span class="st">'palegreen2'</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># here, we apply gordi_predict</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_predict</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>As before, we want to see the name of the predictor variable. Or possibly to change the colour, linewidth, linetype or alpha of the arrow.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span> <span class="fu">gordi_read</span>(<span class="at">env =</span> dune.env,      </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">traits =</span> traits,   </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> T,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">scaling =</span> <span class="st">'species'</span>) <span class="sc">|&gt;</span>  </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">colour =</span> <span class="st">'Subclass'</span>,     </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha =</span> <span class="fl">0.8</span>,             </span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">arrow_size =</span> <span class="fl">0.5</span>,</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">linewidth =</span> <span class="fl">0.5</span>) <span class="sc">|&gt;</span>       </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_colour</span>(<span class="at">scale =</span> <span class="st">'discrete'</span>,    </span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">family =</span> <span class="st">'manual'</span>,        </span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">values =</span> <span class="fu">c</span>(<span class="st">'lightsalmon3'</span>, <span class="st">'palegreen2'</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># here, we apply gordi_predict</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_predict</span>(</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_label =</span> T,         <span class="co"># whether label should be shown</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">repel_label =</span> T,        <span class="co"># should it be repelled?</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">'royalblue3'</span>,  <span class="co"># colour of the arrow (static - if there were more arrows, all would be the same)</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">linewidth =</span> <span class="fl">1.5</span>,        <span class="co"># default = 1</span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">linetype =</span> <span class="dv">4</span>,           <span class="co"># default = 1, for more linetypes see linetype help page</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.7</span>             <span class="co"># opacity of the arrow</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="customise-the-plot" class="level3">
<h3 class="anchored" data-anchor-id="customise-the-plot">6. Customise the plot</h3>
<p>To further customise the ordination plot, we need to do one more thing. We must end the <code>gordi</code> pipeline with <code>gordi_plot()</code> which extracts the ggplot object and makes it available for standard <code>ggplot2</code> functions. From this point, the long <code>pass</code> object, which appears in the Console every time you run some <code>gordi_</code> function is lost.</p>
<blockquote class="blockquote">
<p><strong>Note</strong>: Don’t forget, that traditional <strong>ggplot2</strong> functions use the <code>+</code> sign, not pipe operator <code>|&gt;</code>.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(see)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>m <span class="sc">|&gt;</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">gordi_read</span>(<span class="at">env =</span> dune.env,</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">traits =</span> traits,</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">scaling =</span> <span class="st">'species'</span>,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">correlation =</span> T) <span class="sc">|&gt;</span> </span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_species</span>(<span class="at">colour =</span> <span class="st">'Subclass'</span>) <span class="sc">|&gt;</span> </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_colour</span>(<span class="at">scale =</span> <span class="st">'discrete'</span>, <span class="at">family =</span> <span class="st">'brewer'</span>) <span class="sc">|&gt;</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_predict</span>(<span class="at">show_label =</span> T, <span class="at">repel_label =</span> T, <span class="at">colour =</span> <span class="st">'gold'</span>) <span class="sc">|&gt;</span> </span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gordi_plot</span>() <span class="sc">+</span>      <span class="co"># from here, we are using standard ggplot2 syntax (with "+")</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_abyss</span>() <span class="sc">+</span>     <span class="co"># change default theme</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">'This is a very nice plot'</span>,                        <span class="co"># add custom title and subtitle</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">'Do not try to argue with the authors.'</span>) <span class="sc">+</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fl">0.12</span>, <span class="at">y =</span> <span class="fl">0.15</span>),                            <span class="co"># add another layer (data must be defined manually, no plotting data pass behind the gordi_plot())</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">shape =</span> <span class="dv">8</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">colour =</span> <span class="st">'pink'</span>, <span class="at">stroke =</span> <span class="dv">3</span>)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="gordi_vignette_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="list-of-gordi_-functions" class="level2">
<h2 class="anchored" data-anchor-id="list-of-gordi_-functions">List of <code>gordi_</code> functions</h2>
<ul>
<li><code>gordi_read()</code> - Reads the output from the <code>vegan</code> ordination model and translates it to be used by <code>gordi_</code> functions. <strong>(Must be first in the pipeline)</strong></li>
<li><code>gordi_sites()</code> - Displays <strong>Site</strong> (sample) coordinates (scores) and default labels.</li>
<li><code>gordi_species()</code> - Displays <strong>Species</strong> coordinates (scores) and default labels.</li>
<li><code>gordi_predict()</code> - Displays <strong>Predictor</strong> scores from <strong>constrained</strong> ordinations (like RDA or CCA).</li>
<li><code>gordi_corr()</code> - <strong>Passively fits</strong> environmental variables onto <strong>unconstrained</strong> ordination axes (wrapper for <code>vegan::envfit</code>).</li>
<li><code>gordi_fit()</code> - <strong>Filters species</strong> based on <span class="math inline">\(R^2\)</span> with axes or frequency in the dataset. <strong>Must be placed before <code>gordi_species()</code></strong>.</li>
<li><code>gordi_colour()</code> - Changes default colour palettes when mapped <strong>dynamically</strong>. Must be applied immediately after what it should change.</li>
<li><code>gordi_shape()</code> - Changes default point shapes when mapped <strong>dynamically</strong>. Must be applied immediately after what it should change.</li>
<li><code>gordi_label()</code> - Adjusts labels for species, sites, and predictors. Useful for creating <strong>taxon shortcuts</strong>.</li>
<li><code>gordi_cluster()</code> - Draws <strong>spiders</strong> or <strong>hulls</strong> (polygons) around groups of sites.</li>
<li><code>gordi_plot()</code> - Extracts the plot object for customisation with standard <strong><code>ggplot2</code></strong> syntax.</li>
</ul>
</section>
</section>
<section id="limitations" class="level1">
<h1>Limitations</h1>
<p>Even though <code>gordi_</code> can do a lot of things, there are still many parts of the intended package, which are now under construction. What you may expect it to do but is not possible yet:</p>
<ul>
<li>displaying results of constrained ordination with interactions correctly. There is a discussion about how to calculate the interaction scores correctly, as they are not provided in the output from the <code>scores()</code> function (or at least many of them are missing).</li>
<li>analogue to <code>vegan::ordisurf</code></li>
</ul>
</section>
<section id="contact-the-authors" class="level1">
<h1>Contact the authors</h1>
<p>If you run into a bug or you would welcome some feature, which is not available in <code>gordi</code>, please don’t hesitate to contact the authors:</p>
<ul>
<li>Štěpánka Pustková: 📧 <a href="mailto:499933@muni.cz" class="email">499933@muni.cz</a></li>
<li>Branislav Kopčan: 📧 <a href="mailto:kopcanb@gmail.com" class="email">kopcanb@gmail.com</a></li>
</ul>
<p>You can also contact us anonymously through this <a href="https://forms.gle/KnrFRnYT9jv1JHCC8">Google Form</a>.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>